src/mosaic.mjs

```diff
@@ -217,6 +217,34 @@ async function mosaic512px(z, x, y, filters = {}) {
 
   const tiles = await Promise.all(tilePromises);
 
+  // Begin cloudless stitching logic
+  const filteredTiles = tiles
+    .filter((tile) => !tile.empty())
+    .map((tile, index) => ({
+      tile,
+      meta: metadataByUuid[rows[index].uuid],
+    }));
+
+  // Sort tiles based on the criteria
+  filteredTiles.sort((a, b) => {
+    // 1. Prefer the latest image
+    const dateA = new Date(a.meta.uploaded_at);
+    const dateB = new Date(b.meta.uploaded_at);
+
+    // 2. Prefer higher resolution (lower GSD)
```

Comment:
here we have a nuance: gsd of tile is capped by its zoom level, so if you have a 6 cm/px and 8 cm/px on a tile of zoom 16 they are essentially equal as both will be resampled to 2 m/px. so if you make this capping you can move the logic above the "latest" preference - the images will be showing latest on low zoom and higher resolution when it will start to matter.

Reply:
Fixed